supplier/spring-file-supplier/src/main/java/org/springframework/cloud/fn/supplier/file/FileSupplierConfiguration.java

```diff
@@ -72,6 +75,7 @@ public FileSupplierConfiguration(FileSupplierProperties fileSupplierProperties,
 	}
 
 	@Bean
+	@ConditionalOnExpression("environment['file.supplier.tail'] == null")
```

Comment:
It might simplify this a bit to group the beans that are guarded by the same condition into a static inner configuration class and that way the guard is only written once. Something like...

```java
@Configuration(proxyBeanMethods = false)
@ConditionalOnExpression("environment['file.supplier.tail'] == null")
static class NonTailingConfig {
   ..
}
``` 
That would also simplify the condition on the `fileReadingFlow` bean. 

I don't think its a problem that way it currently written though - just a thought.